src/components/header/index.js

```diff
@@ -0,0 +1,117 @@
+
+        this.state = { 
+            selectedTeamName: null, 
+            selectedTeamTricode: null,
+
+            // NOTE: when the teams are mapped, the raptors are index 27, hence the default number
+            selectedTeam: 27
+        }  
+    }
+
+    componentDidMount() {
+        this.props.getTeams();
+    }
+
+    onSelectChange = (e) => {
+        const { teams, getSelectedTeam } = this.props;
+
+        this.setState({ 
+            selectedTeamName: teams[e.target.value].ttsName, 
+            selectedTeamTricode: teams[e.target.value].tricode,
+            selectedTeam: e.target.value, 
+        });
+
+        getSelectedTeam(teams[e.target.value]);
+    }
+
+    render() {
+        const { selectedTeamTricode, selectedTeam } = this.state;
+        const { teams, selectedTeamColor } = this.props;
+
+        // NOTE: I've used some in-line styling to access the 'selected team color' variable
+
+        return (
+            <div className={styles.headerContainer}>
+                <div className={styles.team} style={{backgroundColor: `${selectedTeamColor}`}}>
+                    <button className={styles.logoContainer}>
+                        <div className={styles.imgContainer}>
+                           <img 
+                                src={selectedTeamTricode ? 
+                                    `https://cdn.nba.net/assets/logos/teams/secondary/web/${selectedTeamTricode}.svg` :
+                                    `https://cdn.nba.net/assets/logos/teams/secondary/web/TOR.svg`
+                                } 
+                                title="team logo" 
+                                alt="team logo"
+                            /> 
+                        </div>
+                        <div className={styles.selectContainer}>
+                            <select 
+                                value={selectedTeam} 
+                                onChange={this.onSelectChange} 
+                                style={{ backgroundColor: `${selectedTeamColor}`}}
+                            >
+                                {teams.map((team, i) => {
+                                    return <option key={i} value={i}>{team.ttsName}</option>;
+                                })}
+                            </select>
+                        </div>
+                    </button>
+                    <div style={{ 
+                            borderStyle: 'solid', 
+                            borderWidth: '45px 20px 0 0', 
+                            borderColor: selectedTeamColor ? `${selectedTeamColor} transparent transparent transparent` :
+                            `#f5f5f5 transparent transparent transparent`,
+                            position: 'absolute',
+                            left: '100%',
+                            zIndex: 99,
+                        }}
+                        
+                        className={styles.borderTriangle}
+                    />
```

Comment:
You only need to make `borderColor` dynamic. Everything else in style can be in the css.

```
<div 
    style={{ borderColor: selectedTeamColor ? 
        `${selectedTeamColor} transparent transparent transparent` :
        `#f5f5f5 transparent transparent transparent`,
     }}
     className={styles.borderTriangle}
/>
```

Reply:
How do you feel about that hard-coded color, #f5f5f5? Is there a way to dynamically render the color variable (ie $grey1)?

Reply:
you can't really use sass variables within js. one of the drawbacks of using css modules. It's fine the way you put it. If you are interested to learn more https://styled-components.com/ . It's pretty cool cause you can use css within js! Mind blown!

Reply:
Let's refactor further:

```
<div 
    style={{ borderColor: `${selectedTeamColor || '#f5f5f5'} transparent transparent transparent` }}
    className={styles.borderTriangle}
/>
```